src/compiler/transformers/module/module.ts

```diff
@@ -1206,7 +1207,31 @@ namespace ts {
                         variables = append(variables, variable);
                     }
                     else if (variable.initializer) {
-                        expressions = append(expressions, transformInitializedVariable(variable as InitializedVariableDeclaration));
+                        if (!isBindingPattern(variable.name) && (isArrowFunction(variable.initializer) || isFunctionExpression(variable.initializer))) {

```

Comment:
Sorry I missed this before, but shouldn't class expressions be included in this as well? They can also be named based on the variable:
```js
export const C = class {};
```

Reply:
No worries, I forgot about them.
I'll update consider them too.

Reply:
Done!